### PR TITLE
fix: fix ffi import external utxo from faucet

### PR DESCRIPTION
**Pull request by hansieodendaal**
Description
---
- Fixed importing external UTXOs via the FFI _by changing the philosophy about having inconsistent recovery byte data in the wallet database_
- Renamed import UTXO methods to be more descriptive
- Added a unit test to test importing external UTXOs via the FFI

_**Note:** The FFI import UTXO method was renamed; this is a breaking change on the FFI interface._

Motivation and Context
---
See above

How Has This Been Tested?
---
Added a new unit test that test the import UTXO interface

### CODE REVIEW

**Review by SWvheerden — DISMISSED** [review]
LGTM

File: base_layer/wallet_ffi/src/lib.rs [hunk 1]
```diff
@@ -5205,11 +5210,11 @@ pub unsafe extern "C" fn wallet_import_utxo(
         return 0;
     }
 
-    if features.is_null() {
-        error = LibWalletError::from(InterfaceError::NullError("features".to_string())).code;
-        ptr::swap(error_out, &mut error as *mut c_int);
-        return 0;
-    }
+    let mut updated_features = if features.is_null() {
```
Comment (stringhandler) [inline base_layer/wallet_ffi/src/lib.rs]: This is wrong. It should error

File: base_layer/wallet_ffi/src/lib.rs [hunk 2]
```diff
@@ -5245,21 +5250,43 @@ pub unsafe extern "C" fn wallet_import_utxo(
 
     let public_script_key = PublicKey::from_secret_key(&(*spending_key));
 
+    // On the blockchain, 'V0' of the output features encode and decode without the recovery byte, whereas 'V1' onwards
+    // adds it in, however, for the wallet, the recovery byte must be consistent with the value commitment to be
+    // accepted in the wallet irrespective if it is 'V0' or 'V1'
+    let recovery_byte =
```
Comment (stringhandler) [inline base_layer/wallet_ffi/src/lib.rs]: This should not be calculated. It should be imported as provided

File: base_layer/wallet_ffi/src/lib.rs [hunk 3]
```diff
@@ -5245,21 +5250,43 @@ pub unsafe extern "C" fn wallet_import_utxo(
...
+                false,
+            )) {
+            Ok(val) => val,
+            Err(e) => {
+                error = LibWalletError::from(OutputManagerError(e)).code;
+                ptr::swap(error_out, &mut error as *mut c_int);
+                return 0;
+            },
+        };
+    updated_features.set_recovery_byte(recovery_byte);
```
Comment (stringhandler) [inline base_layer/wallet_ffi/src/lib.rs]: If you change the data here, the hash will change

**Comment by hansieodendaal** [comment]
Changed the philosophy about having inconsistent recovery byte data in the wallet database

**Review by stringhandler — APPROVED** [review]

**Pull request merged**